File: frontend/src/interfaces/refund_attachment.ts
export interface RefundAttachment {
  ID: number;
  FilePath: string;
  UploadedAt?: string; // เวลาใน Go เป็น time.Time => ใน TS ใช้ string ISO
  RefundID: number;
}

// สำหรับสร้างใหม่ผ่าน API
export interface CreateRefundAttachmentRequest {
  FilePath: string;
  UploadedAt?: string; // optional เวลาสร้างอาจยังไม่ได้ส่ง
  RefundID: number;
}

// สำหรับ backward compatibility หรือการใช้งานเก่า
export interface RefundAttachmentInterface {
  ID: number;
  FilePath: string;
  UploadedAt?: string;
  RefundID: number;
}

File: backend/entity/refund_attachment.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type RefundAttachment struct {
	gorm.Model
	FilePath   string    `json:"file_path"`
	UploadedAt time.Time `json:"uploaded_at"`

	RefundID uint `json:"refund_id"`
}
